### DISCUSSION TITLE
getting started with ggproto extension workshop?

### DISCUSSION THREAD

**Discussion by EvaMaeRey** [ggplot2-extenders/ggplot-extension-club#55] [Q&A]
Given #53 discussion, it seems like breaking this out as a separate topic makes sense.

**Comment by EvaMaeRey** [comment]
ggproto space is super broad. I think Stat focus might make sense there.  I think you can get a whole lot done making Stats and many can be straightforward.  

I think the fact that users use geom_* functions the most, might initially mistakenly direct one to looking for guidance about how to write a Geom, when a Stat might be the solution (one reason layer_* prefixing might be nice).  I'm pretty sure, to some extent, I found that confusing early on.

**Reply by teunbrand** [reply]
Yeah for a 'getting started' workshop, choosing a Stat is probably an early and attainable goal. However, there are basically 2 'slots' that you're supposed to extend, namely `Stat$compute_panel()` and `Stat$compute_group()`. Once you've filled one of these in, I'm unsure what else one can learn about extensions if you're sticking to Stats.

**Reply by EvaMaeRey** [reply]
Yeah, I think Stats focus makes sense and plenty to chew on for newcomers, but advertising as 'ggproto extension workshop' is a big oversell, even if introducing some key moves and concepts about ggproto extension more generally.  I was just kind of going with some of the language in #53...

**Reply by EvaMaeRey** [reply]
... and maybe a series of introductory workshops is what's being implied in #53, cause it's a big space.   😬
